src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -422,6 +422,22 @@ public static async Task<bool> AnyAsync<T>(this ImmutableArray<T> array, Func<T,
             return false;
         }
 
+        public static async Task<bool> AnyAsync<T, TArg>(this ImmutableArray<T> array, Func<T, TArg, Task<bool>> predicateAsync, TArg arg)
+        {
+            int n = array.Length;
+            for (int i = 0; i < n; i++)
+            {
+                var a = array[i];
```

Comment:
Why not basic for each?  It's non allocating for immutable array right?

Reply:
I'm not sure why everything in this file is written this way, but I prefer not to change it for just one case without cause, especially as part of a performance PR.